File: resources/documentation/test-automation/bdd-testing-with-cucumber/steps/loginSteps.js
// steps/loginSteps.js
const { Given, When, Then } = require('@cucumber/cucumber');
const { expect } = require('@playwright/test');
const { LoginPage } = require('../pages/LoginPage');

Given('a registered user with email {string} and password {string}', async function (email, password) {
  this.user = { email, password };
});

When('the user enters their credentials and clicks {string}', async function (buttonName) {
  this.loginPage = new LoginPage(this.page);
  await this.loginPage.login(this.user.email, this.user.password, buttonName);
});

When('the user enters email {string} and password {string}', async function (email, password) {
  this.user = { email, password };
  this.loginPage = new LoginPage(this.page);
  await this.loginPage.enterEmail(email);
  await this.loginPage.enterPassword(password);
});

When('clicks {string}', async function (buttonName) {
  await this.loginPage.clickButton(buttonName);
});

Then('the user should be redirected to {string}', async function (path) {
  await expect(this.page).toHaveURL(new RegExp(`.*${path}$`));
});

Then('the welcome message should display {string}', async function (message) {
  await expect(this.page.locator('[data-testid="welcome-message"]')).toHaveText(message);
});

Then('the login error message should display {string}', async function (message) {
  await expect(this.page.locator('[data-testid="login-error"]')).toHaveText(message);
});

Then('the user should remain on {string}', async function (path) {
  await expect(this.page).toHaveURL(new RegExp(`.*${path}$`));
});

File: resources/documentation/bdd-testing/how-to-write-bdd-scenarios-cucumber/cucumber-jvm/src/test/java/steps/LoginSteps.java
package steps;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import support.LoginPage;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class LoginSteps {
    private final LoginPage loginPage = new LoginPage();

    @Given("I am on the login page")
    public void iAmOnTheLoginPage() {
        loginPage.navigateTo();
    }

    @When("I enter {string} in the email field")
    public void enterEmail(String email) {
        loginPage.enterEmail(email);
    }

    @When("I enter {string} in the password field")
    public void enterPassword(String password) {
        loginPage.enterPassword(password);
    }

    @When("I click the {string} button")
    public void clickButton(String buttonName) {
        loginPage.clickButton(buttonName);
    }

    @When("I log in with SSO")
    public void loginWithSso() {
        loginPage.loginWithSso();
    }

    @Then("I should see {string}")
    public void verifyMessage(String expectedMessage) {
        assertEquals(expectedMessage, loginPage.getMessage());
    }

    @Then("I should be redirected to the dashboard")
    public void verifyDashboardRedirect() {
        assertTrue(loginPage.isDashboardVisible());
    }

    @Then("I should see the dashboard")
    public void verifyDashboard() {
        assertTrue(loginPage.isDashboardVisible());
    }
}
